### ISSUE TITLE
Apollo Tracing changes status code to 500 on error

### ISSUE THREAD

**Issue by oravecjakub** [ChilliCream/graphql-platform#5118]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hot Chocolate returns status code _500 Internal server error_ instead of _400 Bad request_ for invalid or too complex queries when  [Apollo Tracing](https://chillicream.com/docs/hotchocolate/v10/execution-engine/apollo-tracing) feature is enabled Always or OnDemand.
![image](https://user-images.githubusercontent.com/14142226/171842939-d977c7c3-1e73-48e5-a178-c7748b77c1dd.png)

Same behavior was observed when the query exceeded the complexity limit -- 500 with Apollo Tracing, 400 without it.

According to @michaelstaib, tracing should not change the status code https://github.com/ChilliCream/hotchocolate/issues/5116#issuecomment-1145813987

### Steps to reproduce

1. Enable Apollo Tracing feature in On-demand mode (TracingPreference = TracingPreference.OnDemand)
2. Fire invalid request (e.g. request nonexistent field) without GraphQL-Tracing header
3. Observe 400 Bad request status ✔
4. Fire the same request but this time with the GraphQL-Tracing=1 header
5. Observe 500 Internal server error ❌

### Product

Hot Chocolate

### Version

12.11.0-preview.1

**Comment by oravecjakub** [comment]
Thanks for looking into this. I've created an asp.net core project where the issue manifests.
Nugets:
- HotChocolate @ 12.11.0-preview.1
- HotChocolate.AspNetCore @ 12.11.0-preview.1

**Program.cs**
```C#
using HotChocolate.Execution.Options;

var builder = WebApplication.CreateBuilder(args);

builder.Services.AddGraphQLServer()
    .AddApolloTracing(TracingPreference.OnDemand)
    .AddQueryType<QueryType>();

var app = builder.Build();

app.UseRouting();
app.UseEndpoints(endpoints =>
{
    endpoints.MapGraphQL("/graphql");
});

app.Run();


public class QueryType
{
    public string GetHello(string name) => $"Hello, {name}!";
}
```

**Result without X-Apollo-Tracing header**
![image](https://user-images.githubusercontent.com/14142226/171881455-90cc8da2-3d6d-4f03-b43b-63f48e7aa6e3.png)

**Result with X-Apollo-Tracing header**
![image](https://user-images.githubusercontent.com/14142226/171881633-0b5bcd11-feb2-445d-bf33-95a99a4bc9ca.png)

**Source code**
[HC5118_Repro.zip](https://github.com/ChilliCream/hotchocolate/files/8833309/HC5118_Repro.zip)

Hope this helps

**Comment by michaelstaib** [comment]
I have verified the error ... we will do a fix for 12.12.1

**Comment by michaelstaib** [comment]
This one is now fixed for version 12.12.1. I have issued the build and it should be available on NuGet in the next 30 min. I will still keep this issue open until this issue is also fixed for version 13.
